python_modules/dagster/dagster/core/types.py

```diff
@@ -451,8 +451,17 @@ def construct_from_config_value(self, config_value):
         check.failed('should never be called')
 
 
+class DictCounter:
```

Comment:
let's put a comment marking this as `# HACK` and explaining the dagit process reload pitfall